src/logicblocks/event/sources/constraints.py

```diff
@@ -1,6 +1,11 @@
 from abc import ABC
+from collections.abc import Callable
 from dataclasses import dataclass
 
+from logicblocks.event.types import Event
+
+type QueryConstraintCheck[E: Event] = Callable[[E], bool]
```

Comment:
I'm not sure I like that this type has moved. This type is the in-memory adapter specific target type after conversion of a constraint. I don't think it is a general purpose type. Why does it need to move?

Reply:
That's fair

Reply:
Wait, it's needed in `InMemoryEventSource`. So we'd need to change the approach there too

Reply:
It wasn't before though... what changed?